state/remoteapplication.go

```diff
@@ -50,6 +53,29 @@ type remoteEndpointDoc struct {
 	Scope     charm.RelationScope `bson:"scope"`
 }
 
+type attributeMap map[string]interface{}
+
+// RemoteSpace represents a space in another model that endpoints are
+// bound to.
+type RemoteSpace struct {
```

Comment:
Ok - so we'd keep RemoteSpace and RemoteSubnet, but they'd just be what gets returned from RemoteApplication.Spaces(), and remoteSpaceDoc and remoteSubnetDoc (with the same attributes, respectively) would be what's stored in the remoteApplicationDoc?